### PR TITLE
Convert to pytest

### PR DESCRIPTION
**Pull request by UkuLoskit**
There is an issue with the usbprotocol tests, they depend on a particular ordering of the tests. I used pytest-order to preserve the default ordering of unittest. Additionally, these tests cannot be run in parallel. Used this ordering as a crutch to work around this.

### CODE REVIEW

**Comment by maxime-desroches** [comment]
@UkuLoskit Do you plan on finishing this up soon?

**Comment by UkuLoskit** [comment]
@maxime-desroches hey, ready for review now with some caveats described in the PR description.

**Comment by UkuLoskit** [comment]
There are actually more tests ... fixing these

**Comment by adeebshihadeh** [comment]
trigger-jenkins

File: tests/safety/test_elm327.py [hunk 1]
```diff
@@ -26,23 +26,23 @@ def test_tx_hook(self):
...
+      assert should_tx == self._tx(common.make_msg(0, GM_CAMERA_DIAG_ADDR, dat=bytes([byte] * 8)))
 
   def test_tx_hook_on_wrong_safety_mode(self):
     # No point, since we allow many diagnostic addresses
     pass
 
 
 if __name__ == "__main__":
-  unittest.main()
+  pytest.main()
```
Comment (adeebshihadeh) [inline tests/safety/test_elm327.py:48]: the linter should be configured to block this. how is it passing?

File: pyproject.toml [hunk 2]
```diff
@@ -10,6 +10,3 @@ flake8-implicit-str-concat.allow-multiline=false
 
 [tool.ruff.lint.flake8-tidy-imports.banned-api]
 "pytest.main".msg = "pytest.main requires special handling that is easy to mess up!"
-
-[tool.pytest.ini_options]
-addopts = "-n auto"
```
Comment (adeebshihadeh) [inline pyproject.toml:15]: why was this removed? we want parallel by default

File: tests/usbprotocol/test_comms.py [hunk 3]
```diff
@@ -99,31 +102,30 @@ def test_comms_reset_tx(self):
...
+      pkt = libpanda_py.ffi.new('CANPacket_t *')
+      while lpp.can_pop(TX_QUEUES[bus], pkt):
+        queue_msgs.append(unpackage_can_msg(pkt))
 
-          self.assertEqual(len(queue_msgs), len(msgs))
-          self.assertEqual(queue_msgs, msgs)
+      assert len(queue_msgs) == len(msgs)
+      assert queue_msgs == msgs
 
+  @pytest.mark.order(1)
```
Comment (adeebshihadeh) [inline tests/usbprotocol/test_comms.py:128]: why is this needed? the tests should be independent

**Review by adeebshihadeh — CHANGES_REQUESTED** [review]

File: tests/usbprotocol/test_comms.py (hunk 3 above)
Comment (UkuLoskit) [reply tests/usbprotocol/test_comms.py:128]: @adeebshihadeh I have explained this in PR description The tests are dependent on a definite ordering which be reproduced with unittests if you change the method names. E.g test_zzz... for the order(5) five test for example.

Comment (UkuLoskit) [reply tests/usbprotocol/test_comms.py:128]: @adeebshihadeh just bumping this since I haven't heard anything back in a while. I think the easiest way to allow parallelism would be to mark the tests that depend on a particular order to run serially. that way parallelism could still be used for other tests. Fixing the tests themselves, seems like a more complicated task.

**Comment by adeebshihadeh** [comment]
Closing, I left feedback and it wasn't addressed. Feel free to open a new one without any "crutches".
